NineChronicles.Headless/GraphTypes/NodeStatus.cs

```diff
@@ -64,8 +64,36 @@ public NodeStatusType()
                 });
             Field<ListGraphType<TxIdType>>(
                 name: "stagedTxIds",
+                arguments: new QueryArguments(
+                    new QueryArgument<AddressType>
+                    {
+                        Name = "address",
+                        Description = "Target address to query"
+                    }
+                ),
                 description: "Staged TxIds from the current node.",
-                resolve: context => context.Source.BlockChain.GetStagedTransactionIds()
+                resolve: context =>
+                {
+                    if (!context.HasArgument("address"))
+                    {
+                        return context.Source.BlockChain.GetStagedTransactionIds();
+                    }
+                    else
+                    {
+                        Address address = context.GetArgument<Address>("address");
+                        var stagedTransactionIds = context.Source.BlockChain.GetStagedTransactionIds();
+
+                        foreach (var txId in stagedTransactionIds)
+                        {
+                            var tx = context.Source.BlockChain.GetTransaction(txId);
+                            if (tx.Signer != address)
+                            {
+                                stagedTransactionIds = stagedTransactionIds.Remove(txId);
+                            }
+                        }
+                        return stagedTransactionIds;
```

Comment:
These lines could be simpler using [LINQ `.Where()` method][1]:

```suggestion
                        return stagedTransactionIds.Where(txId =>
                            context.Source.BlockChain.GetTransaction(txId).Signer.Equals(address)
                        );
```

[1]: https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.where?view=netstandard-2.0